pkg/daemon/ceph/client/radosnamespace.go

```diff
@@ -0,0 +1,105 @@
+	cmd := NewRBDCommand(context, clusterInfo, args)
+	cmd.JsonOutput = false
+	output, err := cmd.Run()
+	if err != nil {
+		code, ok := exec.ExitStatus(err)
+		if ok && code == int(syscall.EEXIST) {
+			logger.Infof("rados namespace %s/%s already exists", poolName, namespaceName)
+		}
+		return errors.Wrapf(err, "failed to create rados namespace %s/%s. %s", poolName, namespaceName, output)
+
+	}
+
+	logger.Infof("successfully created rados namespace %s/%s", poolName, namespaceName)
+	return nil
+}
+
+func getRadosNamespaceStatistics(context *clusterd.Context, clusterInfo *ClusterInfo, poolName, namespaceName string) (*PoolStatistics, error) {
+	var poolStats PoolStatistics
+
+	args := []string{"pool", "stats", "--pool", poolName, "--namespace", namespaceName}
+	cmd := NewRBDCommand(context, clusterInfo, args)
+	cmd.JsonOutput = true
+	output, err := cmd.Run()
+	if err != nil {
+		code, ok := exec.ExitStatus(err)
+		if ok && code == int(syscall.ENOENT) {
+			return &poolStats, nil
+		}
+		return nil, errors.Wrapf(err, "failed to get pool stats. %s", string(output))
+	}
+	if err := json.Unmarshal(output, &poolStats); err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal pool stats response")
+	}
+
+	return &poolStats, nil
+}
+
+func checkForImagesInRadosNamespace(context *clusterd.Context, clusterInfo *ClusterInfo, poolName, namespaceName string) error {
+	logger.Debugf("checking any images/snapshosts present in pool %s/%s", poolName, namespaceName)
+	stats, err := getRadosNamespaceStatistics(context, clusterInfo, poolName, namespaceName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to list images/snapshosts in pool %s/%s", poolName, namespaceName)
+	}
+	if stats.Images.Count == 0 && stats.Images.SnapCount == 0 {
+		logger.Infof("no images/snapshosts present in pool %s/%s", poolName, namespaceName)
+		return nil
+	}
+
+	return errors.Errorf("pool %s/%s contains %d images and %d snapshosts", poolName, namespaceName, stats.Images.Count, stats.Images.SnapCount)
+}
+
+// DeleteRadosNamespace delete a rados namespace.
+func DeleteRadosNamespace(context *clusterd.Context, clusterInfo *ClusterInfo, poolName, namespaceName string) error {
+	err := checkForImagesInRadosNamespace(context, clusterInfo, poolName, namespaceName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to check if pool %s/%s has rbd images", poolName, namespaceName)
+	}
+	logger.Infof("deleting rados namespace %s/%s", poolName, namespaceName)
+	args := []string{"namespace", "remove", "--pool", poolName, "--namespace", namespaceName}
+	cmd := NewRBDCommand(context, clusterInfo, args)
+	cmd.JsonOutput = false
+	output, err := cmd.Run()
+	if err != nil {
+		code, ok := exec.ExitStatus(err)
+		if !ok || code != int(syscall.ENOENT) {
+			return errors.Wrapf(err, "failed to delete rados namespace %s/%s. %s", poolName, namespaceName, output)
+		}
+	}
+
+	logger.Infof("successfully deleted rados namespace %s/%s", poolName, namespaceName)
```

Comment:
The rook operator can operate on multiple resources in multiple namespaces. All info/debug messages must list the namespaced name so context is clear.

here: cephcluster info